In [16]:
from bs4 import BeautifulSoup
import requests
import pandas as pd
import numpy as np

#  PARLIAMENT CONSTITUENCY

In [4]:
url='https://results.eci.gov.in/PcResultGenJune2024/index.htm'
page=requests.get(url)
soup=BeautifulSoup(page.text,'html')

In [5]:
titles=soup.find_all('th')
titles=[title.text for title in titles]
titles=titles[0:4]
titles

['Party', 'Won', 'Leading', 'Total']

In [6]:
def data(var):
    var=[row.text.strip() for row in var]
    for i in var:
        if i=='':
            var.remove(i)
    return var

In [7]:
dat=[]
for i in range(1,len(soup.find_all('tr'))):
    dat.append(data(soup.find_all('tr')[i]))

In [8]:
len(dat)

43

In [9]:
dat

[['Bharatiya Janata Party - BJP', '240', '0', '240'],
 ['Indian National Congress - INC', '99', '0', '99'],
 ['Samajwadi Party - SP', '37', '0', '37'],
 ['All India Trinamool Congress - AITC', '29', '0', '29'],
 ['Dravida Munnetra Kazhagam - DMK', '22', '0', '22'],
 ['Telugu Desam - TDP', '16', '0', '16'],
 ['Janata Dal  (United) - JD(U)', '12', '0', '12'],
 ['Shiv Sena (Uddhav Balasaheb Thackrey) - SHSUBT', '9', '0', '9'],
 ['Nationalist Congress Party – Sharadchandra Pawar - NCPSP', '8', '0', '8'],
 ['Shiv Sena - SHS', '7', '0', '7'],
 ['Lok Janshakti Party(Ram Vilas) - LJPRV', '5', '0', '5'],
 ['Yuvajana Sramika Rythu Congress Party - YSRCP', '4', '0', '4'],
 ['Rashtriya Janata Dal - RJD', '4', '0', '4'],
 ['Communist Party of India  (Marxist) - CPI(M)', '4', '0', '4'],
 ['Indian Union Muslim League - IUML', '3', '0', '3'],
 ['Aam Aadmi Party - AAAP', '3', '0', '3'],
 ['Jharkhand Mukti Morcha - JMM', '3', '0', '3'],
 ['Janasena Party - JnP', '2', '0', '2'],
 ['Communist Party of Ind

In [10]:
dat.pop(42)

['Total', '543', '0', '543']

In [11]:
df=pd.DataFrame(data=dat,columns=titles)
df.head()

Party  Won Leading Total
0         Bharatiya Janata Party - BJP  240       0   240
1       Indian National Congress - INC   99       0    99
2                 Samajwadi Party - SP   37       0    37
3  All India Trinamool Congress - AITC   29       0    29
4      Dravida Munnetra Kazhagam - DMK   22       0    22

In [12]:
def code(var):
    res=var.split(' ')
    result=res[-1]
    return result
df['Party_Code']=df['Party'].apply(code)

In [13]:
df.head()

Party  Won Leading Total Party_Code
0         Bharatiya Janata Party - BJP  240       0   240        BJP
1       Indian National Congress - INC   99       0    99        INC
2                 Samajwadi Party - SP   37       0    37         SP
3  All India Trinamool Congress - AITC   29       0    29       AITC
4      Dravida Munnetra Kazhagam - DMK   22       0    22        DMK

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Party       42 non-null     object
 1   Won         42 non-null     object
 2   Leading     42 non-null     object
 3   Total       42 non-null     object
 4   Party_Code  42 non-null     object
dtypes: object(5)
memory usage: 1.8+ KB


In [17]:
df['Won']=df['Won'].astype(np.int64)
df['Leading']=df['Leading'].astype(np.int64)
df['Total']=df['Total'].astype(np.int64)

# BJP

In [18]:
url='https://results.eci.gov.in/PcResultGenJune2024/partywisewinresultState-369.htm'
page=requests.get(url)
soup=BeautifulSoup(page.text,'html')

In [19]:
titles=soup.find_all('th')
titles=[title.text.strip() for title in titles]
titles

['S.No',
 'Parliament Constituency',
 'Winning Candidate',
 'Total Votes',
 'Margin']

In [20]:
dat=[]
for i in range(1,len(soup.find_all('tr'))):
    dat.append(data(soup.find_all('tr')[i]))

In [21]:
dat

[['1', 'Anakapalle(5)', 'C.M.RAMESH', '762069', '296530'],
 ['2', 'Rajahmundry(8)', 'DAGGUBATI PURANDHESHWARI', '726515', '239139'],
 ['3',
  'Narsapuram(9)',
  'BHUPATHI RAJU SRINIVASA VARMA (B.J.P.VARMA)',
  '707343',
  '276802'],
 ['4', 'Arunachal West(1)', 'KIREN RIJIJU', '205417', '100738'],
 ['5', 'Arunachal East(2)', 'TAPIR GAO', '145581', '30421'],
 ['6', 'Darrang-Udalguri(4)', 'DILIP SAIKIA', '868387', '329012'],
 ['7', 'Guwahati(5)', 'BIJULI KALITA MEDHI', '894887', '251090'],
 ['8', 'Diphu(6)', 'AMARSING TISSO', '334620', '147603'],
 ['9', 'Karimganj(7)', 'KRIPANATH MALLAH', '545093', '18360'],
 ['10', 'Silchar (8)', 'PARIMAL SUKLABAIDYA', '652405', '264311'],
 ['11', 'Kaziranga(10)', 'KAMAKHYA PRASAD TASA', '897043', '248947'],
 ['12', 'Sonitpur(11)', 'RANJIT DUTTA', '775788', '361408'],
 ['13', 'Lakhimpur(12)', 'PRADAN BARUAH', '663122', '201257'],
 ['14', 'Dibrugarh(13)', 'SARBANANDA SONOWAL', '693762', '279321'],
 ['15', 'Paschim Champaran(2)', 'DR.SANJAY JAISWAL', '5804

In [22]:
bjp=pd.DataFrame(data=dat,columns=titles)
bjp.head()

S.No Parliament Constituency                            Winning Candidate  \
0    1           Anakapalle(5)                                   C.M.RAMESH   
1    2          Rajahmundry(8)                     DAGGUBATI PURANDHESHWARI   
2    3           Narsapuram(9)  BHUPATHI RAJU SRINIVASA VARMA (B.J.P.VARMA)   
3    4       Arunachal West(1)                                 KIREN RIJIJU   
4    5       Arunachal East(2)                                    TAPIR GAO   

  Total Votes  Margin  
0      762069  296530  
1      726515  239139  
2      707343  276802  
3      205417  100738  
4      145581   30421

In [23]:
bjp.drop('S.No',axis=1,inplace=True)

In [24]:
def no_of_candidates(var):
    s=''
    for i in var:
        if i.isdigit():
            s+=i
    return int(s)

In [25]:
bjp['No_of_candidates']=bjp['Parliament Constituency'].apply(no_of_candidates)

In [26]:
bjp.head()

Parliament Constituency                            Winning Candidate  \
0           Anakapalle(5)                                   C.M.RAMESH   
1          Rajahmundry(8)                     DAGGUBATI PURANDHESHWARI   
2           Narsapuram(9)  BHUPATHI RAJU SRINIVASA VARMA (B.J.P.VARMA)   
3       Arunachal West(1)                                 KIREN RIJIJU   
4       Arunachal East(2)                                    TAPIR GAO   

  Total Votes  Margin  No_of_candidates  
0      762069  296530                 5  
1      726515  239139                 8  
2      707343  276802                 9  
3      205417  100738                 1  
4      145581   30421                 2

In [27]:
bjp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 240 entries, 0 to 239
Data columns (total 5 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Parliament Constituency  240 non-null    object
 1   Winning Candidate        240 non-null    object
 2   Total Votes              240 non-null    object
 3   Margin                   240 non-null    object
 4   No_of_candidates         240 non-null    int64 
dtypes: int64(1), object(4)
memory usage: 9.5+ KB


In [28]:
bjp['Total Votes']=bjp['Total Votes'].replace('-',np.nan)

In [29]:
bjp.dropna(inplace=True)

In [30]:
bjp['Total Votes']=bjp['Total Votes'].astype(np.int64)
bjp['Margin']=bjp['Margin'].astype(np.int64)

In [31]:
bjp.info()

<class 'pandas.core.frame.DataFrame'>
Index: 239 entries, 0 to 239
Data columns (total 5 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Parliament Constituency  239 non-null    object
 1   Winning Candidate        239 non-null    object
 2   Total Votes              239 non-null    int64 
 3   Margin                   239 non-null    int64 
 4   No_of_candidates         239 non-null    int64 
dtypes: int64(3), object(2)
memory usage: 11.2+ KB


# INC

In [65]:
def table(url):
    page=requests.get(url)
    soup=BeautifulSoup(page.text,'html')
    titles=soup.find_all('th')
    titles=[title.text.strip() for title in titles]
    dat=[]
    for i in range(1,len(soup.find_all('tr'))):
        dat.append(data(soup.find_all('tr')[i]))
    name=pd.DataFrame(data=dat,columns=titles)
    name.drop('S.No',axis=1,inplace=True)
    def no_of_candidates(var):
        s=''
        for i in var:
            if i.isdigit():
                s+=i
        return int(s)
    try:
        name['No_of_candidates']=name['Parliament Constituency'].apply(no_of_candidates)
    except:
        name['No_of_candidates']=name['Constituency'].apply(no_of_candidates)
    name['Total Votes']=name['Total Votes'].astype(np.int64)
    name['Margin']=name['Margin'].astype(np.int64)
    return name

In [33]:
inc=table('https://results.eci.gov.in/PcResultGenJune2024/partywisewinresultState-742.htm')

In [34]:
inc.head()

Parliament Constituency Winning Candidate  Total Votes   Margin  \
0              Dhubri (2)   RAKIBUL HUSSAIN      1471885  1012476   
1               Nagaon(9)  PRADYUT BORDOLOI       788850   212231   
2              Jorhat(14)      GAURAV GOGOI       751771   144393   
3          Kishanganj(10)    MOHAMMAD JAWED       402850    59692   
4             Katihar(11)       TARIQ ANWAR       567092    49863   

   No_of_candidates  
0                 2  
1                 9  
2                14  
3                10  
4                11

In [35]:
inc.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99 entries, 0 to 98
Data columns (total 5 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Parliament Constituency  99 non-null     object
 1   Winning Candidate        99 non-null     object
 2   Total Votes              99 non-null     int64 
 3   Margin                   99 non-null     int64 
 4   No_of_candidates         99 non-null     int64 
dtypes: int64(3), object(2)
memory usage: 4.0+ KB


# DMK

In [36]:
dmk=table('https://results.eci.gov.in/PcResultGenJune2024/partywisewinresultState-582.htm')
dmk.head()

Parliament Constituency                            Winning Candidate  \
0        CHENNAI NORTH(2)                      DR.KALANIDHI VEERASWAMY   
1        CHENNAI SOUTH(3)  T.SUMATHY (ALIAS) THAMIZHACHI THANGAPANDIAN   
2      CHENNAI CENTRAL(4)                              DAYANIDHI MARAN   
3        SRIPERUMBUDUR(5)                                    T R BAALU   
4         KANCHEEPURAM(6)                                    SELVAM. G   

   Total Votes  Margin  No_of_candidates  
0       497333  339222                 2  
1       516628  225945                 3  
2       413848  244689                 4  
3       758611  487029                 5  
4       586044  221473                 6

In [37]:
dmk.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Parliament Constituency  22 non-null     object
 1   Winning Candidate        22 non-null     object
 2   Total Votes              22 non-null     int64 
 3   Margin                   22 non-null     int64 
 4   No_of_candidates         22 non-null     int64 
dtypes: int64(3), object(2)
memory usage: 1012.0+ bytes


# TDP

In [38]:
tdp=table('https://results.eci.gov.in/PcResultGenJune2024/partywisewinresultState-1745.htm')
tdp.head()

Parliament Constituency         Winning Candidate  Total Votes  Margin  \
0           Srikakulam(2)  KINJARAPU RAMMOHAN NAIDU       754328  327901   
1         Vizianagaram(3)     APPALANAIDU KALISETTI       743113  249351   
2        Visakhapatnam(4)     SRIBHARAT MATHUKUMILI       907467  504247   
3      Amalapuram (SC)(7)     G M HARISH (BALAYOGI)       796981  342196   
4               Eluru(10)        PUTTA MAHESH KUMAR       746351  181857   

   No_of_candidates  
0                 2  
1                 3  
2                 4  
3                 7  
4                10

In [39]:
tdp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16 entries, 0 to 15
Data columns (total 5 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Parliament Constituency  16 non-null     object
 1   Winning Candidate        16 non-null     object
 2   Total Votes              16 non-null     int64 
 3   Margin                   16 non-null     int64 
 4   No_of_candidates         16 non-null     int64 
dtypes: int64(3), object(2)
memory usage: 772.0+ bytes


# MAKING INSIGHTS

In [40]:
import seaborn as sns
import matplotlib.pyplot as plt
from plotly import __version__
from plotly.offline import download_plotlyjs, init_notebook_mode, plot, iplot
import plotly.graph_objs as go
from plotly.subplots import make_subplots
import plotly.express as px
import cufflinks as cf
init_notebook_mode(connected=True)
cf.go_offline()

In [109]:
df.iplot(kind='pie',values='Total',labels='Party_Code')

<Figure size 3000x4000 with 0 Axes>

**This chart shows that BJP has won in 44.2% of the total Parliament Constituencies. The party in the second place is Indian National Congress with 18% of the total Parliament Contituency. The difference in the percentages is large and hence BJP has larger chances to win in the next elections too.**

In [148]:
bjp.iplot(kind='bar',y='No_of_candidates',x='Parliament Constituency')

This shows the number of candidates in each constituency. We can see that Bhadohi has the highest number of candidates

In [149]:
data1=bjp.sort_values('Total Votes').tail(10)
data1.iplot(kind='bar',y='Total Votes',x='Parliament Constituency')

**The Malkajgiri constituency has the highest number of votes for BJP, we can hence conclude that BJP has a higher number of chance of winning in those areas than in Chittorgarh.**

In [181]:
bjp.sort_values('Margin').head(10).iplot(kind='bar',y='Margin',x='Parliament Constituency')

In [182]:
bjp.sort_values('Margin').tail(10).iplot(kind='bar',y='Margin',x='Parliament Constituency')

**These two graphs shows the margin i.e is the difference in votes between the winning party (BJP) and the party in the second place in the ascedning order. This shows that margin in Jaipur is less and hence BJP has more chances winning in Jaipur for the next election, but the margin is the highest in Indore, which means there is less chance of BJP winning in Indore for the next elections.**

In [208]:
df.loc[df['Party_Code']=='INC',:]['Total'].iloc[0]

99

In [209]:
df.loc[df['Party_Code']=='DMK',:]['Total'].iloc[0]

22

In [210]:
sum(df.loc[df['Party_Code']=='INC',:]['Total'],df.loc[df['Party_Code']=='DMK',:]['Total'])

4    121
Name: Total, dtype: int64

In [211]:
df.loc[df['Party_Code']=='BJP',:]['Total'].iloc[0]

240

**The number of winning constituencies for Indian National Congress is very low compared to BJP, but since DMK has allianced with INC, their winning contituencies has raised to 121 which is quite closer to BJP. This shows if INC and DMK continue their collaboration INC have chances of winning the next elections.**

In [212]:
bjp['Total Votes'].sum()

157039540

In [216]:
inc['Total Votes'].sum()

54233901

In [219]:
bjp['Total Votes'].sum()-inc['Total Votes'].sum()

102805639

**This shows the margin of BJP and INC and the difference is very huge which shows that BJP has more power in most of the areas and they are very likely to win the next elections.**

In [41]:
dmk['Parliament Constituency'].unique()

array(['CHENNAI NORTH(2)', 'CHENNAI SOUTH(3)', 'CHENNAI CENTRAL(4)',
       'SRIPERUMBUDUR(5)', 'KANCHEEPURAM(6)', 'ARAKKONAM(7)',
       'VELLORE(8)', 'DHARMAPURI(10)', 'TIRUVANNAMALAI(11)', 'ARANI(12)',
       'KALLAKURICHI(14)', 'SALEM(15)', 'NAMAKKAL(16)', 'ERODE(17)',
       'NILGIRIS(19)', 'COIMBATORE(20)', 'POLLACHI(21)', 'PERAMBALUR(25)',
       'THANJAVUR(30)', 'THENI(33)', 'THOOTHUKKUDI(36)', 'TENKASI(37)'],
      dtype=object)

In [42]:
bjp['Parliament Constituency'].unique()

array(['Anakapalle(5)', 'Rajahmundry(8)', 'Narsapuram(9)',
       'Arunachal West(1)', 'Arunachal East(2)', 'Darrang-Udalguri(4)',
       'Guwahati(5)', 'Diphu(6)', 'Karimganj(7)', 'Silchar (8)',
       'Kaziranga(10)', 'Sonitpur(11)', 'Lakhimpur(12)', 'Dibrugarh(13)',
       'Paschim Champaran(2)', 'Purvi Champaran(3)', 'Madhubani(6)',
       'Araria(9)', 'Darbhanga(14)', 'Muzaffarpur(15)', 'Maharajganj(19)',
       'Saran(20)', 'Ujiarpur(22)', 'Begusarai(24)', 'Patna Sahib(30)',
       'Nawada(39)', 'North Goa(1)', 'Kachchh(1)', 'Patan(3)',
       'Mahesana(4)', 'Sabarkantha(5)', 'Gandhinagar(6)',
       'Ahmedabad East(7)', 'Ahmedabad West(8)', 'Surendranagar(9)',
       'Rajkot(10)', 'Porbandar(11)', 'Jamnagar(12)', 'Junagadh(13)',
       'Amreli(14)', 'Bhavnagar(15)', 'Anand(16)', 'Kheda(17)',
       'Panchmahal(18)', 'Dahod(19)', 'Vadodara(20)',
       'Chhota Udaipur(21)', 'Bharuch(22)', 'Bardoli(23)', 'Navsari(25)',
       'Valsad(26)', 'KURUKSHETRA(2)', 'KARNAL(5)',
       'BH

**It is clearly seen that BJP unlike INC has not won any one constituency in Tamilnadu and Kerala and hence it is very unlikely they would win in Tamilnadu for the next elections.**

In [273]:
fig = make_subplots(rows=2, cols=1)
fig.append_trace(go.Scatter(x=dmk['Parliament Constituency'],y=dmk['Total Votes'],name='Total Votes'), row=1, col=1)
fig.append_trace(go.Scatter(x=dmk['Parliament Constituency'],y=dmk['Margin'],name='Margin'), row=2, col=1)
fig.show()

**From the above graph we can say that, the margin in Dharmapuri, Kallakurichi, Namakkal and Salem are quite low and hence DMK have chances of loosing in these contituencies in the next election.**

In [44]:
df.tail()

Party  Won  Leading  Total  \
37          Aazad Samaj Party (Kanshi Ram) - ASPKR    1        0      1   
38                      Apna Dal (Soneylal) - ADAL    1        0      1   
39                              AJSU Party - AJSUP    1        0      1   
40  All India Majlis-E-Ittehadul Muslimeen - AIMIM    1        0      1   
41                               Independent - IND    7        0      7   

   Party_Code  
37      ASPKR  
38       ADAL  
39      AJSUP  
40      AIMIM  
41        IND

**It is evident that many new entrants like Revolutionary socialist party and AJSU Party were able to win few seats in the election.**

# ASSEMBLY CONSTITUENCY

In [54]:
url='https://results.eci.gov.in/AcResultGenJune2024/partywiseresult-S01.htm'
page=requests.get(url)
soup=BeautifulSoup(page.text,'html')

In [46]:
soup.find_all('table')

[<table class="table">
 <thead>
 <tr>
 <th style="text-align: left;">Party</th>
 <th style="text-align:right">Won</th>
 <th style="text-align:right">Leading</th>
 <th style="text-align:right">Total</th>
 </tr>
 </thead>
 <tbody>
 <tr class="tr">
 <td style="text-align:left">Telugu Desam - TDP</td>
 <td style="text-align:right">
 <a href="partywisewinresult-1745S01.htm">135</a>
 </td>
 <td style="text-align:right">
                                                                                         0
                                                                                     </td>
 <td style="text-align:right">135</td>
 </tr>
 <tr class="tr">
 <td style="text-align:left">Janasena Party - JnP</td>
 <td style="text-align:right">
 <a href="partywisewinresult-860S01.htm">21</a>
 </td>
 <td style="text-align:right">
                                                                                         0
                                                                          

In [55]:
titles=soup.find_all('th')
titles=[title.text for title in titles]
titles=titles[0:4]
titles

['Party', 'Won', 'Leading', 'Total']

In [56]:
dat=[]
for i in range(1,len(soup.find_all('tr'))):
    dat.append(data(soup.find_all('tr')[i]))

In [57]:
dat.pop(-1)

['Total', '175', '0', '175']

In [58]:
andhra=pd.DataFrame(data=dat,columns=titles)
andhra

Party  Won Leading Total
0                             Telugu Desam - TDP  135       0   135
1                           Janasena Party - JnP   21       0    21
2  Yuvajana Sramika Rythu Congress Party - YSRCP   11       0    11
3                   Bharatiya Janata Party - BJP    8       0     8

In [59]:
andhra['Party_Code']=andhra['Party'].apply(code)

In [62]:
andhra

Party  Won Leading Total Party_Code
0                             Telugu Desam - TDP  135       0   135        TDP
1                           Janasena Party - JnP   21       0    21        JnP
2  Yuvajana Sramika Rythu Congress Party - YSRCP   11       0    11      YSRCP
3                   Bharatiya Janata Party - BJP    8       0     8        BJP

In [63]:
andhra.iplot(kind='pie',values='Total',labels='Party_Code')

**We can clearly see the power of Telugu Desam(TDP) in the general election to assembly constituencies and BJP has secured the least number of seats in this region.**

In [66]:
ap=table('https://results.eci.gov.in/AcResultGenJune2024/partywisewinresult-1745S01.htm')

In [67]:
ap.head()

Constituency      Winning Candidate  Total Votes  Margin Status  \
0  Ichchapuram(1)         ASHOK BENDALAM       110612   39783  22/22   
1       Palasa(2)        GOUTHU SIREESHA       101560   40350  21/21   
2      Tekkali(3)  ATCHANNAIDU KINJARAPU       107923   34435  24/24   
3  Pathapatnam(4)     MAMIDI GOVINDA RAO        89452   26527  24/24   
4   Srikakulam(5)          GONDU SHANKAR       117091   52521  20/20   

   No_of_candidates  
0                 1  
1                 2  
2                 3  
3                 4  
4                 5

In [68]:
bp=table('https://results.eci.gov.in/AcResultGenJune2024/partywisewinresult-369S01.htm')
bp.head()

Constituency                                  Winning Candidate  \
0              Etcherla(7)                              ESWARA RAO NADUKUDITI   
1  Visakhapatnam North(23)                         VISHNU KUMAR RAJU PENMETSA   
2            Anaparthy(40)                       RAMAKRISHNA REDDY NALLAMILLI   
3             Kaikalur(73)                                  KAMINENI SRINIVAS   
4      Vijayawada West(79)  YALAMANCHILI SATYANARAYANA CHOWDARY (SUJANA CH...   

   Total Votes  Margin Status  No_of_candidates  
0       112770   29089  23/23                 7  
1       108801   47534  21/21                23  
2       105720   20850  17/17                40  
3       109280   45273  17/17                73  
4       105669   47032  20/20                79

In [77]:
ap[ap['Total Votes']==ap['Total Votes'].max()]

Constituency    Winning Candidate  Total Votes  Margin Status  \
16  Bhimili(20)  GANTA SRINIVASA RAO       176230   92401  26/26   

    No_of_candidates  
16                20

In [71]:
ap['Total Votes'].sum()

14658528

In [72]:
bp['Total Votes'].sum()

848353

In [73]:
ap['Total Votes'].sum()-bp['Total Votes'].sum()

13810175

**The margin of Telugu Desam and BJP is not even close and hence BJP has very less chances of winning in this area and GANTA SRINIVASA RAO has the power in that area.**

In [80]:
url='https://results.eci.gov.in/AcResultGenJune2024/partywiseresult-S18.htm'
page=requests.get(url)
soup=BeautifulSoup(page.text,'html')

In [81]:
titles=soup.find_all('th')
titles=[title.text for title in titles]
titles=titles[0:4]
titles

['Party', 'Won', 'Leading', 'Total']

In [82]:
dat=[]
for i in range(1,len(soup.find_all('tr'))):
    dat.append(data(soup.find_all('tr')[i]))
dat.pop(-1)

['Total', '147', '0', '147']

In [83]:
od=pd.DataFrame(data=dat,columns=titles)
od

Party Won Leading Total
0                  Bharatiya Janata Party - BJP  78       0    78
1                         Biju Janata Dal - BJD  51       0    51
2                Indian National Congress - INC  14       0    14
3  Communist Party of India  (Marxist) - CPI(M)   1       0     1
4                             Independent - IND   3       0     3

In [84]:
od['Party_Code']=od['Party'].apply(code)

In [85]:
od

Party Won Leading Total Party_Code
0                  Bharatiya Janata Party - BJP  78       0    78        BJP
1                         Biju Janata Dal - BJD  51       0    51        BJD
2                Indian National Congress - INC  14       0    14        INC
3  Communist Party of India  (Marxist) - CPI(M)   1       0     1     CPI(M)
4                             Independent - IND   3       0     3        IND

**Here it is evident that BJP is leading with 78 seats. Hence we can conclude that BJP has more power over the north part of India than the southern part of India.**